Homework_4/Repositories/Repository.cs

```diff
@@ -0,0 +1,15 @@
+using Homework_4.Provider;
+
+namespace Homework_4.Repositories;
+
+public abstract class Repository
+{
+    private string ConnectionString { get; }
+    protected PurchaseDbContext DbProvider { get; }
```

Comment:
this is just DbContext and not provider, I'd better call this class 'provider' even though it's far from being it